### PR TITLE
[PB-250]: feature/add global search

### PR DESCRIPTION
**Pull request by rafijv**
- Support global searching across all files and folders within the user's Drive, regardless of their location or hierarchy.
- Allow to filter by type, folders, pdf...
- Go to the folder/file item on clicking one result
- Breadcrumbs change on clicking one result

### CODE REVIEW

File: src/app/core/components/Navbar/Navbar_New_SearchBar.tsx [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { StorageFilters } from '../../../store/slices/storage/storage.model';
 import { sessionSelectors } from '../../../store/slices/session/session.selectors';
 import { Workspace } from '../../types';
 import { UserSettings } from '@internxt/sdk/dist/shared/types/userSettings';
+import { SearchResult } from '@internxt/sdk/dist/drive/storage/types';
```
Comment (CandelR) [inline src/app/core/components/Navbar/Navbar_New_SearchBar.tsx:7]: I know that you haven't created the component but, could you change the name of the file from snake_case to camelCase? For consistency with other components

File: src/app/core/components/Navbar/Navbar_New_SearchBar.tsx [hunk 2]
```diff
@@ -28,20 +34,32 @@ interface NavbarProps {
...
+const isSelectedType = (extension: string, extensionMap: FileExtensionMap) => {
+  for (const fileType in extensionMap) {
+    if (extensionMap[fileType].includes(extension.toLowerCase())) {
+      return true;
+    }
+  }
+  return false;
+};
+
 // TODO: SUBSTITUE THE NAVBAR WITH THIS ONE WITH NEW SEARCH BAR WHEN BACKED IS READY
```
Comment (CandelR) [inline src/app/core/components/Navbar/Navbar_New_SearchBar.tsx]: if it is already substituted remove this TODO.

File: src/app/core/components/Navbar/Navbar_New_SearchBar.tsx [hunk 3]
```diff
@@ -71,26 +91,34 @@ const Navbar = (props: NavbarProps) => {
...
 
   useEffect(() => {
-    search();
+    if (filters.length > 0) {
+      const filteredSearchResults = searchResult.filter((result) => {
+        for (const filter of filters) {
+          if (filter === 'folder' && result.itemType?.toLowerCase() === 'folder') {
+            return true;
+          }
+          if (result.item.type && isSelectedType(result.item.type, fileExtension[filter || 'default'])) {
```
Comment (CandelR) [inline src/app/core/components/Navbar/Navbar_New_SearchBar.tsx]: Maybe extracting the conditionals to const would be more readable.
For example:
`const isFolder = filter === 'folder' && result.itemType?.toLowerCase() === 'folder'`
`const hasAllowedFileType = result.item.type && isSelectedType(result.item.type, fileExtension[filter || 'default'])`

Also extracting the logic to an arrow function.

File: src/app/core/components/Navbar/Navbar_New_SearchBar.tsx [hunk 4]
```diff
@@ -107,11 +135,20 @@ const Navbar = (props: NavbarProps) => {
     }
   };
 
-  const openItem = (id: string) => {
-    searchInput.current?.blur();
-    // TODO -> Open file/folder by id
-    // Open file/folder in the same tab
-    alert(`Open item with ID ${id}`);
+  const openItem = (item) => {
+    if (item.itemType === 'folder' || item.itemType === 'FOLDER') {
```
Comment (CandelR) [inline src/app/core/components/Navbar/Navbar_New_SearchBar.tsx]: Could it only be that two possibilities? 
Perhaps using `item.itemType.toLowerCase() === 'folder' `might save us some future trouble.

File: src/app/core/config/app.json [hunk 5]
```diff
@@ -76,7 +76,8 @@
       "layout": "header-and-sidenav",
       "path": "/app/recents",
       "exact": true,
-      "auth": true
+      "auth": true,
+      "hideSearch": true
```
Comment (CandelR) [inline src/app/core/config/app.json:80]: Does the old search have to disappear in other Views?

File: src/app/core/layouts/HeaderAndSidenavLayout/HeaderAndSidenavLayout.tsx [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import { Link, useLocation } from 'react-router-dom';
 
-import Navbar from '../../components/Navbar/Navbar';
+import Navbar from '../../components/Navbar/Navbar_New_SearchBar';
+// import Navbar from '../../components/Navbar/Navbar';
```
Comment (CandelR) [inline src/app/core/layouts/HeaderAndSidenavLayout/HeaderAndSidenavLayout.tsx]: You missed the old import comment.

**Review by CandelR — COMMENTED** [review]
I did a quick test and the global search does not find the folders. I understand that it should find them, right?

File: src/app/core/components/Navbar/Navbar_New_SearchBar.tsx (hunk 1 above)
Comment (rafijv) [reply src/app/core/components/Navbar/Navbar_New_SearchBar.tsx:7]: Done

File: src/app/core/components/Navbar/Navbar_New_SearchBar.tsx (hunk 2 above)
Comment (rafijv) [reply src/app/core/components/Navbar/Navbar_New_SearchBar.tsx]: Done

File: src/app/core/components/Navbar/Navbar_New_SearchBar.tsx (hunk 3 above)
Comment (rafijv) [reply src/app/core/components/Navbar/Navbar_New_SearchBar.tsx]: Done

File: src/app/core/components/Navbar/Navbar_New_SearchBar.tsx (hunk 4 above)
Comment (rafijv) [reply src/app/core/components/Navbar/Navbar_New_SearchBar.tsx]: Done

File: src/app/core/layouts/HeaderAndSidenavLayout/HeaderAndSidenavLayout.tsx (hunk 6 above)
Comment (rafijv) [reply src/app/core/layouts/HeaderAndSidenavLayout/HeaderAndSidenavLayout.tsx]: Done

File: src/app/core/config/app.json (hunk 5 above)
Comment (rafijv) [reply src/app/core/config/app.json:80]: The search bar was displayed in two other views, recent and shared but neither of them worked, and the global search only works in drive so I have hidden it.

**Review by CandelR — APPROVED** [review]

**Pull request merged**